homeassistant/components/motion_blinds/__init__.py

```diff
@@ -205,10 +216,14 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
     )
 
     if unload_ok:
+        multicast = hass.data[DOMAIN][KEY_MULTICAST_LISTENER]
+        multicast.Unregister_motion_gateway(config_entry.data[CONF_HOST])
         hass.data[DOMAIN].pop(config_entry.entry_id)
 
-    if len(hass.data[DOMAIN]) == 1:
+    if len(hass.data[DOMAIN]) == 3:
```

Comment:
This check feels a bit fragile. We may add something to `hass.data[DOMAIN]` and forget to update here. Maybe check the number of loaded config entries for the domain instead. If there is only one loaded config entry for the domain, we unsubscribe and stop.

Reply:
Is there a function (probably as part of hass.) to get all loaded config entries of a domain?

Reply:
I think I got it: `hass.config_entries.async_entries(DOMAIN)`

Reply:
Tested and it works, thanks for the tip!